File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/mad/Java/C++/s452855095_comments_removed.cpp
#include <iostream>
#include <vector>

int main() {
    std::ios::sync_with_stdio(false);
    std::cin.tie(nullptr);

    int n;
    std::cin >> n;
    std::vector<int> arr(n);
    for(int i = 0; i < n; i++)
        std::cin >> arr[i];
    
    int cnt = 0;
    bool flag = true;
    while(flag) {
        flag = false;
        for(int i = 0; i < n; i++) {
            if(arr[i] % 2 != 0) {
                flag = true;
                break;
            } else {
                arr[i] /= 2;
            }
        }
        if(!flag)
            cnt++;
    }
    std::cout << cnt;
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/mad/Java/Go/s452855095_comments_removed.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	n, err := strconv.Atoi(scanner.Text())
	if err != nil {
		fmt.Println("Error reading input:", err)
		return
	}
	scanner.Scan()
	input := strings.Fields(scanner.Text())
	arr := make([]int, n)
	for i := 0; i < n; i++ {
		var err error
		arr[i], err = strconv.Atoi(input[i])
		if err != nil {
			fmt.Println("Error reading input:", err)
			return
		}
	}
	cnt := 0
	flag := true
	for {
		flag = true
		for i := 0; i < n; i++ {
			if arr[i]%2 != 0 {
				flag = false
				break
			} else {
				arr[i] /= 2
			}
		}
		if flag {
			cnt++
		} else {
			break
		}
	}
	fmt.Println(cnt)
}
// End of Code
